### PR TITLE
Saga slice story api

### PR DESCRIPTION
**Pull request by t-samborski**
I created a Sage Slice and Story that handle the Popular Movies page state. The page loads for a short time, then the repositories from the API are downloaded (not all of them yet), and in case of an error, an error icon is displayed.

### CODE REVIEW

File: src/features/MoviesBrowser/movieBrowserSaga.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { takeLatest, put, call } from "redux-saga/effects";
+import { fetchRepositories, fetchRepositoriesSuccess, fetchRepositoriesError } from "./movieBrowserSlice";
+import { getRepositories } from "./moviesBrowserAPI";
+
+
+
+function* fetchRepositoriesHendler() {
```
Comment (KingaGrelewicz) [inline src/features/MoviesBrowser/movieBrowserSaga.js:7]: why repositories? we want to get movies no repositories

File: src/features/MoviesBrowser/PopularMovies/Status/index.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import { Error } from "./Error";
+import { Loading } from "./Loading";
+import { Sucess } from "./Succes";
+
+export const Content = ({status, repositories}) => {
```
Comment (KingaGrelewicz) [inline src/features/MoviesBrowser/PopularMovies/Status/index.js:5]: I like this solution :)

File: src/core/store.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {configureStore} from "@reduxjs/toolkit";
+import movieBrowserReducer from '../features/MoviesBrowser/movieBrowserSlice';
+import createSagaMiddleware from "redux-saga";
+import saga from "./saga";
+
+const sagaMiddleware = createSagaMiddleware();
+
+const store = configureStore({
+    reducer: {
+        movieBrowser: movieBrowserReducer,
```
Comment (KingaGrelewicz) [inline src/core/store.js:10]: You can add this reducer to store.js which already exist

File: src/features/MoviesBrowser/moviesBrowserAPI.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import axios from "axios";
+
+// const APIkey = "https://api.themoviedb.org/3/movie/11?api_key=f577b54446e4b39433952b4514de7d17";
```
Comment (KingaGrelewicz) [inline src/features/MoviesBrowser/moviesBrowserAPI.js:3]: I think you can delete commented part if don't use

File: src/features/MoviesBrowser/movieBrowserSaga.js (hunk 1 above)
Comment (t-samborski) [reply src/features/MoviesBrowser/movieBrowserSaga.js:7]: Out of habit, I assumed they were film repositories. If we don't like it in the final stage, we can change the name. :)

File: src/features/MoviesBrowser/PopularMovies/Status/index.js (hunk 2 above)
Comment (t-samborski) [reply src/features/MoviesBrowser/PopularMovies/Status/index.js:5]: ;)

File: src/core/store.js (hunk 3 above)
Comment (t-samborski) [reply src/core/store.js:10]: Of course

File: src/features/MoviesBrowser/moviesBrowserAPI.js (hunk 4 above)
Comment (t-samborski) [reply src/features/MoviesBrowser/moviesBrowserAPI.js:3]: Thanks, I must have missed that

File: src/features/MoviesBrowser/movieBrowserSaga.js (hunk 1 above)
Comment (KingaGrelewicz) [reply src/features/MoviesBrowser/movieBrowserSaga.js:7]: Yes, I think popularMovies or just movies would be better. We can change it in the following week :)

**Review by KingaGrelewicz — APPROVED** [review]

**Review by KingaGrelewicz — APPROVED** [review]

**Pull request closed without merging**